framework/src/mesh/MooseMesh.C

```diff
@@ -3699,3 +3702,22 @@ MooseMesh::lengthUnit() const
   mooseAssert(_coord_transform, "This must be non-null");
   return _coord_transform->lengthUnit();
 }
+
+void
+MooseMesh::checkDuplicateSubdomainNames()
+{
+  std::map<SubdomainName, SubdomainID> subdomain;
+  for (const auto & sbd_id : _mesh_subdomains)
+  {
+    std::string sub_name = getSubdomainName(sbd_id);
+    if (!sub_name.empty() && subdomain.count(sub_name) > 0)
+      mooseError("The subdomain name ",
+                 sub_name,
+                 " has already been used for the subdomain with ID=",
+                 subdomain[sub_name],
+                 ", Please rename the subdomain with ID=",
+                 sbd_id);
```

Comment:
let s reword this to be "is used by both subdomains with ID". ... " and ID " ...
Please rename one of them).

we dont know which one is the best to rename